src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2450,8 +2450,27 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatement(SyntaxKind par
                     }
 
                     var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
+
+                    var containsErrorDiagnostic = false;
+                    // ContainsDiagnostics will be true if *any child node* contains 
+                    // a Diganostic, so only 
                     if (incompleteMember.ContainsDiagnostics)
```

Comment:
>incompleteMember.ContainsDiagnostics [](start = 24, length = 36)

We should check other instances where this API is used in the parser and even in binding, I see some that look like candidates for the same fix. This should be done for VB as well. Also, It looks like we already have a way to enumerate all diagnostics for green nodes, but it is implemented for VB only, see implementation of ```Friend Overridable Function GetSyntaxErrors() As IList(Of DiagnosticInfo)``` in ```Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode```. It looks like that implementation is all in terms of GreenNode, so we can simply move that implementation to the GreenNode itself and rename the method to GetSyntaxDiagnostics because the current name is somewhat misleading. Also, there is no reason for that method to return IList(Of DiagnosticInfo), IEnumerable(Of DiagnosticInfo) would work just fine. Then we can add a method called HasErrors to the GreenNode and implement it either by reusing GetSyntaxDiagnostics, or by reusing parts of logic from it (if we want to avoid creating an enumerator). This place should simply use the HasErrors method from GreenNode.